pkg/clockcache/cache.go

```diff
@@ -174,7 +174,7 @@ func (self *Cache) evict() (insertPtr *element) {
 			}
 
 			if insertPtr != nil {
-				self.next = next + 1
+				self.next = (startLoc + next + 1) % self.Len()
```

Comment:
are we sure we don't need `% self.Len()` on line 189 too? Even if startLoc is the last element?

Reply:
Assuming that `self.storage` has length 10, and `startLoc == 9` (i.e. it points to the last element).

When we slice `self.storage` into `postStart` and `preStart`, the contents of `postStart` will be one item, the last item, and the contents of `preStart` will be the first 9 items.

This is because go's slices are half-open (i.e. exclusive of the last element).